packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -46,14 +46,79 @@ export async function createNewFluidFile(
     getStorageToken: (options: TokenFetchOptions, name: string) => Promise<string | null>,
     newFileInfo: INewFileInfo,
     logger: ITelemetryLogger,
-    createNewSummary: ISummaryTree,
+    createNewSummary: ISummaryTree | undefined,
     epochTracker: EpochTracker,
 ): Promise<IOdspResolvedUrl> {
     // Check for valid filename before the request to create file is actually made.
     if (isInvalidFileName(newFileInfo.filename)) {
         throwOdspNetworkError("Invalid filename. Please try again.", invalidFileNameStatusCode);
     }
 
+    const itemId = createNewSummary === undefined
+        ? await createNewEmptyFluidFile(getStorageToken, newFileInfo, logger, epochTracker)
+        : await createNewFluidFileFromSummary(getStorageToken, newFileInfo, logger, createNewSummary, epochTracker);
+
+    const odspUrl = createOdspUrl({... newFileInfo, itemId, dataStorePath: "/"});
+    const resolver = new OdspDriverUrlResolver();
+    return resolver.resolve({ url: odspUrl });
+}
+
+export async function createNewEmptyFluidFile(
+    getStorageToken: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    newFileInfo: INewFileInfo,
+    logger: ITelemetryLogger,
+    epochTracker: EpochTracker,
+): Promise<string> {
+    const filePath = newFileInfo.filePath ? encodeURIComponent(`/${newFileInfo.filePath}`) : "";
+    // remove .fluid extension for empty files
+    const extensionlessFileName = newFileInfo.filename.endsWith(".fluid")
```

Comment:
And other than in fluid-fetch tool, I do not see any place where we ever use .fluid. So I'd avoid using it here as well, leaving it up to host and UX layer on top of storage (SPO libraries / front-end) to define what kind of extensions map to fluid apps.
Is the intention here to create a file that does not map to fluid app temporarily? If so, I'd rather add (.tmp) to it, then to remove it, as removing assumes you know extension when in regality we do not.